### ISSUE TITLE
exported not working within jquery (but works outside of it)

### ISSUE THREAD

**Issue by kkmuffme** [eslint/eslint#11173]
* **ESLint Version:** v4.19.1
* **Node Version:** v7.10.1
* **npm Version:** 4.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```
{
    "env": {
        "browser": true,
        "worker": true,
        "phantomjs": true,
        "qunit": true,
        "jquery": true,
        "serviceworker": true,
        "es6": true
    },
    "extends": "eslint:recommended",
}
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
jQuery( function() {
        /* exported abcd */
        var abcd = 'abc';
        var xyz = 'xyz';
});

// same result when I try this:
/* exported abcd */
jQuery( function() {
        var abcd = 'abc';
        var xyz = 'xyz';
});
```

```bash
eslint --report-unused-disable-directives --fix file.js
```

**What did you expect to happen?**
```
  4:6  error  'xyz' is assigned a value but never used   no-unused-vars
```

**What actually happened?**
```
  3:6  error  'abcd' is assigned a value but never used  no-unused-vars
  4:6  error  'xyz' is assigned a value but never used   no-unused-vars
```

**Are you willing to submit a pull request to fix this bug?**
Unfortunately I have zero idea how to implement a fix, but I can offer any help besides developing (QA, documentation,...)
